Updated Data_validation Code

breadsp opened the pull request:
updated code, removed other files since are blank
Trimmed code to run only prior_clinical, demographic, and biospecimen

Will add other files in later once these are in good shape

breadsp committed:
updated data validation base code.  Updated file-object code.  Added in new file for equipment/consumable/reagent/aliquot

breadsp committed:
remove other_files

These files are not ready to use just yet

breadsp committed:
Updated and cleaned files

Tweaked code some to reduce duplicate function calls, updated looping logic per Ming's suggestion regarding indexes

breadsp committed:
Updated files based on Codacy feedback

Cleaned up files based on Codacy recomendations
removed blank spaces and unused variables (will be needed in future version)

breadsp committed:
Changes from Codacy

Updated Prior_Test_Results to include Duration type (Day/Month/Year) validation rule

breadsp committed:
Updating all Files

n2iw commented:
- file_validator_object.py: there are quite some "&" and "|" used in the code, please double check they meant to be bit operator instead of logic "and" and "or". Especially in line 158, 167 and 169.
- prior_test_result_validator.py, please double check if '|' meant to be bit operator or logic operator "or" in line 79 and 85.

breadsp committed:
Updated Files, fixed logical issues

Fixed some logical issues between & and "and" usage
Added the other-valiadator which includes Aliquot,equipment,reagent, and consumable
